Controller/ResourceController.php

```diff
@@ -32,21 +34,34 @@ public function __construct(
 
     public function resourceAction($repositoryName, $path)
     {
-        $repository = $this->registry->get($repositoryName);
-        $resource = $repository->get('/' . $path);
-
-        $context = SerializationContext::create();
-        $context->enableMaxDepthChecks();
-        $context->setSerializeNull(true);
-        $json = $this->serializer->serialize(
-            $resource,
-            'json',
-            $context
-        );
-
-        $response = new Response($json);
-        $response->headers->set('Content-Type', 'application/json');
-        
+        try {
+            $repository = $this->registry->get($repositoryName);
+            $resource = $repository->get('/' . $path);
+
+            $context = SerializationContext::create();
+            $context->enableMaxDepthChecks();
+            $context->setSerializeNull(true);
+            $json = $this->serializer->serialize(
+                $resource,
+                'json',
+                $context
+            );
+
+            $response = new Response($json);
+            $response->headers->set('Content-Type', 'application/json');
+        } catch (\Exception $e) {
```

Comment:
In that case, can't we just require FOSRestBundle and configure it using the prepended extension technique? FOSRestBundle already has this logic.

When moving to FOSRestBundle, we can also make it do other things like the serialization stuff.